types/config.go

```diff
@@ -16,22 +16,21 @@ type Config struct {
 	bech32AddressPrefix map[string]string
 	txEncoder           TxEncoder
 	addressVerifier     func([]byte) error
-	mtx                 sync.RWMutex
 	coinType            uint32
-	sealed              bool
```

Comment:
This was there to prevent the structure to be modified further once the method `Seal()` had been called and was just complementary to the (wrongly managed) mutex (which your changes fix). Can you reintroduce this and the `Seal()` method please?

Reply:
Why do we need the seal now if it is safe to work with?

Reply:
I guess the original design was to prevent further modifications to be carried out once one has called the method `Seal()`. goroutine safety is orthogonal to sealing